pkg/target/http.go

```diff
@@ -399,55 +399,60 @@ func (ht *HTTPTarget) Write(messages []*models.Message) (*models.TargetWriteResu
 			}
 
 			// Ensure response body is always closed
-			if resp.StatusCode >= 200 && resp.StatusCode < 300 {
-				// Drain and close the body for successful responses
-				_, _ = io.Copy(io.Discard, resp.Body)
-				resp.Body.Close()
-
-				for _, msg := range goodMsgs {
-					if msg.AckFunc != nil { // Ack successful messages
-						msg.AckFunc()
+			func() {
+				defer resp.Body.Close()
```

Comment:
Perhaps safer as it guarantees `resp.Body.Close()`

Reply:
As far as I understand, deferring guarantees the close in that it reduces the possibility of a panic, by waiting until the enclosing function has returned.

I think that if you don't defer, but instead ensure it's only ever called once the code is done with it - it's still just as safe.

But that's something I'll need to validate before I'm totally confident.